zipkin-lens/src/slices/remoteServicesSlice.ts

```diff
@@ -77,6 +77,4 @@ const remoteServicesSlice = createSlice({
   },
 });
 
-export const { clearRemoteServices } = remoteServicesSlice.actions;
```

Comment:
Thanks!
It looks like these codes can be removed as well :)

https://github.com/openzipkin/zipkin/blob/5562dd5b3677944027450f546fe4b9cc1b2ca5bd/zipkin-lens/src/slices/remoteServicesSlice.ts#L55-L59

https://github.com/openzipkin/zipkin/blob/5562dd5b3677944027450f546fe4b9cc1b2ca5bd/zipkin-lens/src/slices/spansSlice.ts#L55-L59

https://github.com/openzipkin/zipkin/blob/5562dd5b3677944027450f546fe4b9cc1b2ca5bd/zipkin-lens/src/slices/spansSlice.ts#L80

Reply:
great

Reply:
I am curious on these.. why can I remove them? Maybe you can tell me for each below:

servicesSlice, autocompleteKeys - only loaded ever once, so no need to clear
autocompleteValues - currently no reducer; data is partitioned by 'autocompleteKey'
remoteServices, spans - reducer to remove; data partitioned by 'service'
dependencies - contains a clear reducer, the data is not really cache-able except what's defined in HTTP

Later I would like to refactor based on this conversation. I would like to delete code, just I don't understand when we need a reducer to clear and not. cc @anuraaga

Reply:
https://github.com/openzipkin/zipkin/blob/5562dd5b3677944027450f546fe4b9cc1b2ca5bd/zipkin-lens/src/components/DiscoverPage/SearchBar/SearchBar.tsx#L138-L149

As soon as service criterion is changed, the spans and remoteServices fetching will begin, and the previous spans and remoteServices will be cleared.
So I think there is no need to explicitly clear them.

Reply:
thanks!